event/event.go

```diff
@@ -134,18 +134,22 @@ func (s *Subscription[T]) Serve(handler Handler[T]) error {
 	return s.rawsub.Serve(SampledMessageHandler(func(msg Message) error {
 		var event Envelope[T]
 
+		ctx := context.Background()
+		log := slog.FromCtx(ctx)
+
 		if err := json.Unmarshal(msg.Body(), &event); err != nil {
+			log.Error("unable to parse event as JSON", "error", err, "event", msg)
 			return fmt.Errorf("parsing event as JSON, event: %v, error: %v", msg, err)
 		}
 
 		if event.Name != s.name {
+			log.Error("event name doesn't match handler", "event_expected", s.name, "event_received", event.Name)
```

Comment:
```suggestion
			log.Error("event name doesn't match handler", "expected", s.name, "received", event.Name)
```

Given that the message is already clear about it being the even name.

Reply:
Yeah, I wasn't sure about it